xds/src/test/java/io/grpc/xds/internal/security/certprovider/FileWatcherCertificateProviderProviderTest.java

```diff
@@ -48,13 +53,32 @@ public class FileWatcherCertificateProviderProviderTest {
       scheduledExecutorServiceFactory;
   @Mock private TimeProvider timeProvider;
 
+  @Parameter
+  public Boolean enableSpiffe;
+  private Boolean originalEnableSpiffe;
   private FileWatcherCertificateProviderProvider provider;
 
+  @Parameters(name = "enableSpiffe={0}")
+  public static Collection<Boolean> data() {
+    return ImmutableList.of(true, false);
+  }
+
   @Before
   public void setUp() throws IOException {
     provider =
         new FileWatcherCertificateProviderProvider(
             fileWatcherCertificateProviderFactory, scheduledExecutorServiceFactory, timeProvider);
+    saveEnvironment();
+    FileWatcherCertificateProviderProvider.enableSpiffe = enableSpiffe;
+  }
+
+  private void saveEnvironment() {
```

Comment:
You don't need this separate `saveEnvironment` method. Just inline it in setUp().